src/frontend/sharedComponents/Images/TrulyContainedImage.tsx

```diff
@@ -12,32 +15,28 @@ export function TrulyContainedImage({
   onLoad,
   style,
   ...baseImageProps
-}: Omit<
-  ImageProps,
-  // Resize mode has no effect.
-  | 'resizeMode'
-  // Dimensions are dictated by the container that lives in the consuming component.
-  | 'height'
-  | 'width'
->) {
+}: Omit<ExpoImageProps, 'contentFit'>) {
   const [aspectRatio, setAspectRatio] = useState<number | undefined>(undefined);
 
   return (
-    <Image
+    <ExpoImage
       {...baseImageProps}
-      resizeMode="contain"
+      contentFit="contain"
       style={[
         style,
         // There seems to be a bug related to images that have non-network sources (e.g. `file://...`)
         // where the loading the image won't even be attempted unless the image has a known height prior.
         // We trigger the load attempt by initially setting the height to a non-zero value and then removing it once it loads.
-        {aspectRatio, height: aspectRatio ? undefined : 1},
+        // Note: with React Native's image, the initial height can be set to 1 but with Expo's Image, it needs to be a larger value or else
+        // there's some layout glitchiness each time it loads...
+        {
+          aspectRatio,
+          height: aspectRatio ? undefined : 50,
+        },
       ]}
       onLoad={event => {
```

Comment:
What if instead of exposing the onLoadEvent callback we expose an `onImageStorageSizeLoaded` event that returns the image size, where internally inside this component we do `getExpoImageStorageSize`. It would just simplify this component a bit, and really the consuming component does not need any other on load info

Reply:
I think it's helpful to expose the onLoad as-is because the need for size information is usage-specific. As a shared component, this ideally could be used anywhere and I don't know if having a specific prop about getting the size of the image is strictly necessary in the other potential cases.

Open to updating this component if it becomes needed more often elsewhere but otherwise think it's an unnecessary change for now.